miden-lib/asm/sat/internal/faucet.masm

```diff
@@ -0,0 +1,261 @@
+use.std::collections::smt
+
+use.miden::sat::internal::account
+use.miden::sat::internal::asset
+use.miden::sat::internal::asset_vault
+use.miden::sat::internal::layout
+
+# FUNGIBLE ASSETS
+# ==================================================================================================
+
+#! Mints a fungible asset associated with the fungible faucet the transaction is being executed
+#! against.
+#!
+#! Panics:
+#! - If the transaction is not being executed against a fungible faucet.
+#! - If the fungible asset being minted is not associated with the faucet the transaction is
+#!   being executed against.
+#! - If the asset is not well formed.
+#! - If the total issuance after minting is greater than the maximum amount allowed.
+#!
+#! Stack: [ASSET]
+#! Outputs: [ASSET]
+#!
+#! - amount is the amount of the fungible asset to mint.
+#! - ASSET is the asset that was minted.
+export.mint_fungible_asset
+    # assert the fungible asset is well formed
+    exec.asset::validate_fungible_asset
+    # => [ASSET]
+
+    # assert that the asset is associated with the faucet the transaction is being executed against
+    exec.account::get_id exec.asset::validate_fungible_asset_origin
+    # => [ASSET]
```

Comment:
Optional nit: I'd probably run `validate_fungible_asset` as a part of `validate_fungible_asset_origin` and then here we'd just need to do a single call to `validate_fungible_asset_origin`. But I can see how this approach could be beneficial as well - so, don't really have a strong opinion on this.

(but if we do this for fungible assets, we should do the same for non-fungible assets as well).

Reply:
Happy to go with your proposal.  On this note, I think we should do a review of the procedures in `miden::sat::internal::*`.  I think it could be valuable to make the part of the user facing api (`midne::sat::*`).